### ISSUE TITLE
CSS-in-JS not working with a babel configuration present

### ISSUE THREAD

**Issue by ivancuric** [stylelint/stylelint#4018]
> Clearly describe the bug
Using stylelint in a CSS-in-JS context results in it not working without a custom processor, specifically `stylelint-processor-styled-components`. With it on, it works as intended.

Discussion on twitter: https://twitter.com/_baxuz/status/1116364962086232067

> Which rule, if any, is the bug related to?

None

> What CSS is needed to reproduce the bug?
Any, but for instance:

```css
.demo {
    font-size: 16px;
    font-size: 16px;
}
.demo {
    font-size: 16px;
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "extends": ["stylelint-config-standard", "stylelint-config-prettier"],
}
```

> Which version of stylelint are you using?

9.10.1

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI, with `npx stylelint demo.jsx` and VScode

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

no

> What did you expect to happen?

Warnings of any kind

> What actually happened (e.g. what warnings or errors did you get)?

No warnings were reported.

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

Would you mind creating a repository with reproducible problem, please? I'm trying your example in [demo](https://stylelint.io/demo/) and can't reproduce the issue. Maybe it's due `.jsx` extension.

![Screenshot 2019-04-11 at 21 15 48](https://user-images.githubusercontent.com/654597/55985106-59f08080-5c9f-11e9-8476-09ad4bf3ece1.png)

**Comment by ivancuric** [comment]
Will do tommorow. I've tried with all extensions: `.js`, `.jsx`, `.ts`, and `.tsx`.

**Comment by hudochenkov** [comment]
That's a weird one, because I use styled components linting and autofixing everyday at work. We'll wait for your repo :)

**Comment by ivancuric** [comment]
Ok this is weird. A new repo with the exact same items in `package.json` and `.stylelintrc` don't break. Will follow up when I get more info.

**Comment by ivancuric** [comment]
Ha-ha! It's `babel.config.js` that messes it up:
```js
module.exports = function(api) {
  api.cache(true);
  return {
    presets: [
      ['@babel/preset-env', { targets: { node: 'current' } }],
      '@babel/preset-react',
      '@babel/preset-typescript',
    ],
    plugins: [
      '@babel/plugin-syntax-dynamic-import',
      '@babel/proposal-class-properties',
      '@babel/proposal-object-rest-spread',
    ],
  };
};
```

**Comment by samuelkraft** [comment]
I'm having the exact same issue, deleting `.babelrc` makes it work

**Comment by jeddy3** [comment]
@ivancuric Thanks for digging deeper and for uncovering that the problem lies with the interplay between stylelint and Babel. The postcss parser ([postcss-jsx](https://github.com/gucong3000/postcss-jsx)) that stylelint relies on to extract the styles from containers like styled-components uses babel under the hood. I suspect the issue might be to do with postcss-jsx and some recent change in Babel.

I'll relabel this "investigation needed", any help by anyone to look into this would be appreciated!

**Comment by otofu-square** [comment]
Found workaround for this ... 😑 

```
// babel.config.js
if (process.env.NODE_ENV !== "stylelint") {
  module.exports = {
    presets: [...],
    plugins: [...],
  };
}

// package.json
  "scripts": {
    ...,
    "stylelint": "NODE_ENV=stylelint stylelint --syntax css-in-js './src/**/*.{ts,tsx}'"
  },
```

**Comment by hudochenkov** [comment]
Repository with reproducible problem would help to investigate this issue. I use stylelint with styled components, and I have `.babelrc.js` in my project as well, but don't have any issues.

**Comment by ivancuric** [comment]
https://github.com/ivancuric/stylelint-bug
Here you go.

**Comment by trainiac** [comment]
@hudochenkov how is your setup different such that you are able to make things work?  I'm seeing the same issue with a `stylelint.config.js` and `babel.config.js` in the root of my project.

**Comment by hudochenkov** [comment]
I figured out the issue.

`babel.config.js` is [project-wide configuration](https://babeljs.io/docs/en/config-files#project-wide-configuration). It interferes with Babel in [`postcss-jsx`](https://github.com/gucong3000/postcss-jsx), package that we use to work with CSS-in-JS.

I use `.babelrc.js`, that's why CSS-in-JS works for me. So if you don't need features of `babel.config.js`, convert it to `.babelrc.js`.

To fix the problem complete changes should be made to `postcss-jsx`.

**Comment by trainiac** [comment]
Thanks @hudochenkov. I changed the filename from `babel.config.js` to `.babelrc.js` and I still had the problem.  The only way I can workaround it is to put all of my work in its own parallel folder to node_modules.  

```
project/
   node_modules/
   mycode/
     .babelrc.js
```

**Comment by billyvg** [comment]
I've found that `--syntax css-in-js` is not compatible with `stylelint-processor-styled-components` - is there any overlap between the two? which one should we be using?

**Comment by evilebottnawi** [comment]
Your don't need `stylelint-processor-styled-components` anymore, stylelint support this format out of box

**Comment by hudochenkov** [comment]
We've updated dependency, the issue should be resolved. Until we release new version of stylelint you can reinstall stylelint to have `postcss-jsx` updated to the latest version. Run `npm ls postcss-jsx` to check version. It should be 0.36.2.

**Comment by ivancuric** [comment]
Thanks!
